CORE_DEVELOPER.txt

```diff
@@ -0,0 +1,164 @@
+contributor, point them to our development guide, or other GitHub
+workflow tutorials around the web. Do not assume that they know how
+GitHub works (e.g., many don't realize that adding a commit
+automatically updates a pull request). Gentle, polite, kind
+encouragement can make the difference between a new core developer or
+an abandoned pull request.
+
+When reviewing, focus on the following:
+
+1. **API**: The API is what users see when they first use
+   `scikit-image`. APIs are difficult to change once released, so
+   should be  simple, `functional
+   <https://en.wikipedia.org/wiki/Functional_programming>`__ (i.e. not
+   carry state), consistent with other parts of the library, and
+   should avoid modifying input variables.
+
+2. **Documentation:** Any new feature should have a gallery
+   example, that not only illustrates but explains it.
+
+3. **The algorithm:** You should understand the code being modified or
+   added before approving it.  (See “if you merge it, you buy it”,
+   below.) Implementations should do what they claim, and be simple,
+   readable, and efficient.
+
+4. **Tests:** All contributions to the library *must* be tested.  Good
+   tests not only execute the code, but explores corner cases.  It is
+   tempting not to review tests, but please do so.
+
+Other changes may be *nitpicky*: spelling mistakes, formatting,
+etc. Do not ask contributors to make these changes, and instead
+make the changes by `pushing to their branch
+<https://help.github.com/articles/committing-changes-to-a-pull-request-branch-created-from-a-fork/>`__ [enabling-push-to-branch]_,
+or using GitHub’s `suggestion
+<https://help.github.com/articles/commenting-on-a-pull-request/>`__
+`feature
+<https://help.github.com/articles/incorporating-feedback-in-your-pull-request/>`__.
+(The latter is preferred because it gives the contributor a choice in
+whether to accept the changes.)
+
+.. [enabling-push-to-branch]: https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/
+
+Unless you know that a contributor is experienced with git, don’t
+ask for a rebase when merge conflicts arise. Instead, rebase the
+branch yourself and force-push to their branch.  If the contributor is
+no longer active, you may take over their branch by submitting a new pull
+request.
+
+Please add a note to a PR after you push new changes; GitHub does not
+send out notifications for these.
+
+Merge Only Changes You Understand
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*Long-term maintainability* is an important concern.  Code doesn't
+merely have to *work*, but should be *understood* by multiple core
+developers.  Changes will have to be made in the future, and the
+original contributor may have moved on.
+
+Therefore, *do not merge a code change unless you understand
+it*. Instead, ask for help! It is no shame for a core developer to ask
+a community member, or even an external developer, for added insight.
```

Comment:
This is the only bit of your changes that I take issue with. I dislike bringing the word "shame" into this, even as a negation. Let's emphasise the positives! For reference, here was my text for this bit:

> scikit-image has a long history of core developers asking contributors or even external developers for help understanding things. Each case is an opportunity to grow as an individual and as a team.

I would very much like to keep the second sentence, and some modification of the first one to emphasise that this is the standard way of approaching things.